src/material-experimental/mdc-checkbox/harness/checkbox-harness.ts

```diff
@@ -0,0 +1,128 @@
+/**
+ * @license
+ * Copyright Google LLC All Rights Reserved.
+ *
+ * Use of this source code is governed by an MIT-style license that can be
+ * found in the LICENSE file at https://angular.io/license
+ */
+
+import {ComponentHarness, HarnessPredicate} from '@angular/cdk-experimental/testing';
+import {coerceBooleanProperty} from '@angular/cdk/coercion';
+
+/**
+ * Harness for interacting with a standard mat-checkbox in tests.
+ * @dynamic
+ */
+export class MatCheckboxHarness extends ComponentHarness {
+  static hostSelector = 'mat-checkbox';
+
+  /**
+   * Gets a `HarnessPredicate` that can be used to search for a checkbox with specific attributes.
+   * @param options Options for narrowing the search:
+   *   - `label` finds a checkbox with specific label text.
+   * @return a `HarnessPredicate` configured with the given options.
+   */
+  static with(options: {label?: string | RegExp} = {}): HarnessPredicate<MatCheckboxHarness> {
+    return new HarnessPredicate(MatCheckboxHarness)
+        .addOption('label', options.label,
+            (harness, label) => HarnessPredicate.stringMatches(harness.getLabelText(), label));
+  }
+
+  private _label = this.locatorFor('.mat-checkbox-label');
+  private _input = this.locatorFor('input');
+
+  /** Gets a boolean promise indicating if the checkbox is checked. */
+  async isChecked(): Promise<boolean> {
+    const checked = (await this._input()).getAttribute('checked');
+    return coerceBooleanProperty(await checked);
+  }
+
+  /** Gets a boolean promise indicating if the checkbox is in an indeterminate state. */
+  async isIndeterminate(): Promise<boolean> {
+    const indeterminate = (await this._input()).getAttribute('indeterminate');
+    return coerceBooleanProperty(await indeterminate);
+  }
+
+  /** Gets a boolean promise indicating if the checkbox is disabled. */
+  async isDisabled(): Promise<boolean> {
+    const disabled = (await this._input()).getAttribute('disabled');
+    return coerceBooleanProperty(await disabled);
+  }
+
+  /** Gets a boolean promise indicating if the checkbox is required. */
+  async isRequired(): Promise<boolean> {
+    const required = (await this._input()).getAttribute('required');
+    return coerceBooleanProperty(await required);
+  }
+
+  /** Gets a boolean promise indicating if the checkbox is valid. */
+  async isValid(): Promise<boolean> {
+    const invalid = (await this.host()).hasClass('ng-invalid');
+    return !(await invalid);
+  }
+
+  /** Gets a promise for the checkbox's name. */
+  async getName(): Promise<string | null> {
+    return (await this._input()).getAttribute('name');
+  }
+
+  /** Gets a promise for the checkbox's value. */
+  async getValue(): Promise<string | null> {
```

Comment:
Omit this since it should be the same as `isChecked`?

Reply:
It's not the same, its the `value` attribute on the `<input>`